### ISSUE TITLE
Add computed_diagnostic data type to data schema

### ISSUE THREAD

**Issue by djgagne** [NCAR/miles-credit#350]
I want to add a new data category for diagnostic variables that are computed by a postblock and are included in the loss calculation, like geopotential or mean sea level pressure. These variables are not output by the model directly, but we do want to incorporate them in the training loss to encourage the model to enforce physical consistency. The new category is `computed_diagnostic`. 

For variables that we want to compute later but not include in the training, like fields on pressure levels, we can just define postblocks that can be added at inference time to the config file. Maybe we also include an inference_only flag for those postblocks so they are skipped during training?
